src/Neo/Ledger/Blockchain.cs

```diff
@@ -480,6 +484,45 @@ private void Persist(Block block)
                 Debug.Assert(header.Index == block.Index);
         }
 
+        internal static async Task InvokeCommittingAsync(NeoSystem system, Block block, DataCache snapshot, IReadOnlyList<ApplicationExecuted> applicationExecutedList)
+        {
+            await InvokeHandlersAsync(Committing?.GetInvocationList(), h => ((CommittingHandler)h)(system, block, snapshot, applicationExecutedList));
+        }
+
+        internal static async Task InvokeCommittedAsync(NeoSystem system, Block block)
+        {
+            await InvokeHandlersAsync(Committed?.GetInvocationList(), h => ((CommittedHandler)h)(system, block));
+        }
+
+        private static async Task InvokeHandlersAsync(Delegate[] handlers, Action<Delegate> handlerAction)
+        {
+            if (handlers == null) return;
+
+            var exceptions = new ConcurrentBag<Exception>();
+            var tasks = handlers.Select(handler => Task.Run(() =>
+            {
+                try
+                {
+                    handlerAction(handler);
+                }
+                catch (Exception ex) when (handler.Target is Plugin)
```

Comment:
we need this when? it's only used for plugins

Reply:
yes, need to check plugin exceptions seperately.